### PR TITLE
[libusb] Fix usage

### LINKED ISSUE #35558: CMake Cannot Find Libusb Header Files
**Describe the bug**
When I downloaded and installed libusb using Vcpkg, I found that I couldn't locate the header files for Libusb.
```cmake
find_package(libusb CONFIG REQUIRED)
target_include_directories(main PRIVATE ${LIBUSB_INCLUDE_DIRS})
target_link_libraries(main PRIVATE ${LIBUSB_LIBRARIES})
```

**Environment**
- OS: Windows
- Compiler: MSVC 2019

**To Reproduce**
```bash
./vcpkg install [xxxx](libusb:x64-windows-static)
cmake .. -DCMAKE_TOOLCHAIN_FILE=path/to/vcpkg/scripts/buildsystems/vcpkg.cmake \
              -DVCPKG_TARGET_TRIPLET=x64-windows-static
```

**Expected behavior**

```cmake
message(${LIBUSB_INCLUDE_DIR})
# LIBUSB_INCLUDE_DIR-NOTFOUND
```

### PR DESCRIPTION
**Pull request by MonicaLiu0311**
Fixes #35480. Use `libusb` pkgconfig,  remove `vcpkg-cmake-wrapper.cmake`.

The usage test passed on `x64-windows` (header files found):
```
libusb can be imported via CMake FindPkgConfig module:
    find_package(PkgConfig REQUIRED)
    pkg_check_modules(libusb REQUIRED IMPORTED_TARGET libusb-1.0)

    target_link_libraries(main PRIVATE PkgConfig::libusb)
```

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] ~~SHA512s are updated for each updated download~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/libusb/usage [hunk 1]
```diff
@@ -1,5 +1,6 @@
-libusb provides CMake targets:
+libusb can be imported via CMake FindPkgConfig module:
+    find_package(PkgConfig REQUIRED)
+    pkg_check_modules(LIBUSB REQUIRED IMPORTED_TARGET libusb-1.0)
 
-    find_package(libusb CONFIG REQUIRED)
-    target_include_directories(main PRIVATE ${LIBUSB_INCLUDE_DIRS})
+    target_include_directories(main PRIVATE ${LIBUSB_INCLUDE_DIRS}) 
     target_link_libraries(main PRIVATE ${LIBUSB_LIBRARIES})
```
Comment (dg0yt) [inline ports/libusb/usage:6]:
```suggestion
    target_link_libraries(main PRIVATE PkgConfig::LIBUSB)
```

**Comment by dg0yt** [comment]
While I do recommend to suggest the imported target from pkg_check_modules for immediate downstream usage, it implies that immediate consumers must `pkg_check_modules` etc. to their exported CMake config. And also generated pc files must be checked to not list the target in `Libs:`.

**Comment by MonicaLiu0311** [comment]
> it implies that immediate consumers must `pkg_check_modules` etc

Please help confirm: https://github.com/microsoft/vcpkg/pull/35498/commits/b7199fd1017cf4cc1397fba16acc5a35ee545b8d


> And also generated pc files must be checked to not list the target in `Libs:`.

It has been checked locally.

**Comment by MonicaLiu0311** [comment]
Feature `libusb` is tested successfully in the following ports:
```
pcl[libusb] 
freerdp[urbdrc]
```

`gazebo[libusb]` is not tested due to errors in its dependency `dartsim` which will not be resolved in this PR:
```
G:\vcpkg\buildtrees\dartsim\src\v6.12.2-05a3588b9d.clean\python\dartpy\gui\osg\GUIEventHandler.cpp(34): fatal error C1083: Cannot open include file: 'dart/gui/osg/osg.hpp': No such file or directory
```

**Review by FrankXie05 — APPROVED** [review]

**Comment by pedroCX486** [comment]
I can't integrate this as a dependency after this PR, this broke my project.

**Comment by MonicaLiu0311** [comment]
> I can't integrate this as a dependency after this PR, this broke my project.

Please provide detailed error information.

**Pull request merged**